mmedit/datasets/pipelines/augmentation.py

```diff
@@ -145,6 +148,53 @@ def __repr__(self):
         return repr_str
 
 
+@PIPELINES.register_module()
+class RandomRotation:
+    """Rotate the image by a randomly-chosen angle, measured in degree.
+
+    Args:
+        keys (list[str]): The images to be rotated.
+        degrees (tuple[float] | tuple[int] | float | int): If it is a tuple,
+        it represents a range (min, max). If it is a float or int,
+        the range is constructed as (-degrees, degrees).
+    """
+
+    def __init__(self, keys, degrees):
+        if isinstance(degrees, float) or isinstance(degrees, int):
+            if degrees < 0.0:
+                raise ValueError(
+                    'If degrees is a single float number, it must be positive.'
+                )
+            else:
+                degrees = (-degrees, degrees)
+        elif mmcv.is_tuple_of(degrees, float) or mmcv.is_tuple_of(
+                degrees, int):
+            degrees = degrees
+        elif degrees is not None:
+            raise TypeError(f'Degrees must be float | int or tuple of float | '
+                            'int, but got '
+                            f'{type(degrees)}.')
+
+        self.keys = keys
+        self.degrees = degrees
+
+    def __call__(self, results):
+        angle = random.uniform(self.degrees[0], self.degrees[1])
+
+        for k in self.keys:
+            results[k] = mmcv.imrotate(results[k], angle)
+            if results[k].ndim == 2:
+                results[k] = np.expand_dims(results[k], axis=2)
```

Comment:
I think we can keep the same number of dimensions in case someone wants to simply rotate the image without adding an additional dimension.

Reply:
Keeping the same number of dimensions might cause error later when broadcasting image and mask before training. (think I have encountered that error before...)

Reply:
I dont think so, all my input images are rgb, but I'll try recreate the error again and see what's wrong

Reply:
Oh I think a few images in places365 dataset are grayscale images, hence the error

Reply:
I think we can keep this, as we probably need the channel dimension, even for a grayscale image.